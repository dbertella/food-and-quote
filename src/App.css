*, :after, :before {
    box-sizing: border-box;
}

.App-header {
  background-color: #222;
  padding: 20px;
  color: white;
}

.App-intro {
  font-size: large;
}

.container {
  max-width: 640px;
  width: 98%;
  margin: 0 auto;
  overflow-x: hidden;
}
h2 {
  text-align: center;
}
img {
  max-width: 100%;
  height: auto;
}

button {
  font-size: 17px;
  margin: 2px;
  border: 1px solid #333;
  background: transparent;
  color: #333;
}
button:hover {
  background: #f3f3f3;
}

input {
  width: 100%;
  height: 50px;
  font-size: 2em;
  border: 2px solid #333;
}
