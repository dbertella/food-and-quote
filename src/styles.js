export const PRIMARY = "#31f6a9";

export const SECONDARY = "#373737";

export const TEXT_COLOR = SECONDARY;