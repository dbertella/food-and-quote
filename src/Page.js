// @flow
import React, { Component } from 'react';
import { connect } from 'react-redux';
import Helmet from "react-helmet";
import styled from "styled-components";

import BackButton from './components/BackButton';
import Container from './components/Container';
import Loader from './components/Loader';
import * as actions from './actions';
import { createMarkup } from './utils';
import { TEXT_COLOR } from './styles';
import type { Post } from './types';

export const TitleWrap = styled.div`
  display: flex;
  align-items: center;
`;
export const Title = styled.h1`
  color: ${TEXT_COLOR};
  font-size: 17px;
`;

class Page extends Component {
  props: {
    post: Post,
    match: Object,
    requestPostById: Function,
    isFetching: boolean,
    history: Object,
  }

  componentDidMount() {
    const { match, post, requestPostById } = this.props;
    if (!post.title) {
      requestPostById(match.params.id);
    }
  }

  render() {
    const {
      post,
      isFetching,
      history,
    } = this.props;
    if (isFetching) {
      return <Loader />
    }
    return (
      <Container>
        <Helmet
          title={post.title}
          meta={[
              {property: "og:type", content: "article"}
          ]}
          link={[
              {rel: "canonical", href: post.URL},
          ]}
        />
        {
          post.featured_image &&
          <img src={`${post.featured_image}?w=640&h=360&crop=1`} alt={post.title} />
        }
        <TitleWrap>
          <BackButton onClick={history.goBack} />
          <Title dangerouslySetInnerHTML={createMarkup(post.title)} />
        </TitleWrap>
        <div dangerouslySetInnerHTML={createMarkup(post.content)} />
      </Container>
    )
  }
}

const mapStateToProps = (state, ownProps) => {
  const post = {
    ...state.post[ownProps.match.params.id]
  };
  return ({
    post,
    isFetching: state.post.isFetching,
  })
};

export default connect(
  mapStateToProps, {
  requestPostById: actions.requestPostById,
})(Page);
