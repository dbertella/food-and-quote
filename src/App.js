import React from "react";
import {
  Route,
  Link,
} from "react-router-dom";
import Helmet from "react-helmet";
import styled from "styled-components";
import List from "./List";
import Page from "./Page";
import Search from "./Search";
import "./App.css";
import logo from './img/icon.png';

const Title = styled(Link)`
  color: #fff;
  text-decoration: none;
  padding: 20px;
`;
const AppHeader = styled.div`
  display: flex;
  alignItems: center;
  background-color: #222;
  padding: 20px;
`;
const App = () => (
  <div className="App">
      <Helmet
        htmlAttributes={{lang: "en", amp: undefined}}
        titleTemplate="%s | Food and quote"
        defaultTitle="Food and quote - 3 ingredient recipes app"
        titleAttributes={{itemprop: "name", lang: "en"}}
        meta={[
            {name: "description", content: "Food and quote is a 3 ingredient recipes food blog with parenting quotes. Find easy recipes for the family and share your experience."},
            {property: "og:type", content: "article"}
        ]}
        link={[
            {rel: "canonical", href: "https://foodandquote.com"},
        ]}
    />
    <AppHeader>
      <img src={logo} alt="Food and quote logo" />
      <h1><Title to="/">Food and Quote</Title></h1>
    </AppHeader>
    <div>
      <Route exact path="/" component={Search} />
      <Route path="/s" component={List} />
      <Route path="/recipe/:id" component={Page} />
    </div>
  </div>
);

export default App;
