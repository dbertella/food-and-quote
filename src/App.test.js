import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import { TitleWrap, Title } from './Page';
import renderer from 'react-test-renderer';
import { TEXT_COLOR } from './styles';

jest.mock('react-router', () => 'Route');

test('TitleWrap renders correctly', () => {
  const tree = renderer.create(
    <TitleWrap />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

test('Title renders correctly', () => {
  const tree = renderer.create(
    <Title />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

// it('renders without crashing', () => {
//   const div = document.createElement('div');
//   ReactDOM.render(<App />, div);
// });


