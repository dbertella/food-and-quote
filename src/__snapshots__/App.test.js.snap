exports[`test Title renders correctly 1`] = `
<h1
  className="cflcSu" />
`;

exports[`test TitleWrap renders correctly 1`] = `
<div
  className="dqUvXD" />
`;
